### PR TITLE
CLOUDP-225800: Sign AtlasCLI docker images with Cosign

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-225800

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
     runs-on: ubuntu-latest
     env:
-      IMAGE_REPOSITORY: mongodb/atlas
+      STAGING_IMAGE_REPOSITORY: mongodb/apix_test
       PLATFORMS: linux/amd64,linux/arm64
-      QUAY: quay.io
     steps:
       - name: Check out code
-        uses: actions/checkout@v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
```
Comment (blva) [inline .github/workflows/docker-release.yml:16]: [q] why do we need to pin the version to commit shas across this file?

Comment (andreaangiolillo) [reply .github/workflows/docker-release.yml:16]: This is a good article to get more info: https://www.paloaltonetworks.com/blog/prisma-cloud/github-actions-worm-dependencies/

TLDR: 

> Pin actions using a commit hash to reduce the risk of using a maliciously modified action.
